spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerOperations.java

```diff
@@ -37,6 +37,23 @@
  */
 public interface SpannerOperations {
 
+	/**
+	 * Gets the DDL string to create the table for the given entity in Spanner. This is just one of
+	 * the possible schemas that can support the given entity type. The specific schema is determined
+	 * by the configured property type converters used by the read and write methods in
+	 * this SpannerOperations and will be compatible with those methods.
+	 * @param entityClass the entity type.
+	 * @return the DDL string.
+	 */
+	String getCreateTableDDLString(Class entityClass);
```

Comment:
I think this feature should go together with introducing a `SpannerAdminTemplate` that wraps the `DatabaseAdminClient` and then these could go on that template instead of `SpannerOperations`.

Reply:
At first I thought so too, but generating just the *string* i dont think is a database admin client responsibility. I think the database admin's responsibility is running this string and actually creating the table. This is supported by:

- generating the schema requires extensive use of the converter
- requires extensive access to the underlying entities
- requires use of converter utils

All of these things would be strange to have inside just database admin class. I think the admin template should accept these strings, but not be in charge of actually generating them.

Reply:
Hmmm...maybe we should just create a separate class for it, with the right collaborators, e.g. `MappingSchemaGenerator`, it really doesn't belong to the DML world.

Reply:
sounds good. coming right up!